pkg/querier/worker/worker.go

```diff
@@ -123,23 +123,33 @@ func newQuerierWorkerWithProcessor(cfg Config, log log.Logger, processor process
 		processor: processor,
 	}
 
-	w, err := util.NewDNSWatcher(address, cfg.DNSLookupPeriod, f)
-	if err != nil {
-		return nil, err
+	// Empty address is only used in tests, where individual targets are added manually.
+	if address != "" {
+		w, err := util.NewDNSWatcher(address, cfg.DNSLookupPeriod, f)
+		if err != nil {
+			return nil, err
+		}
+
+		servs = append(servs, w)
 	}
 
-	servs = append(servs, w)
+	if len(servs) > 0 {
+		subservices, err := services.NewManager(servs...)
```

Comment:
Should we make the manager work with an empty list of services? Would simplify the logic in the worker.

Reply:
I'd prefer not to complicate manager's logic more than it already is.